bzip2-sys/build.rs

```diff
@@ -31,6 +31,16 @@ fn main() {
         .flag_if_supported("-Wl,--gc-sections")
         .flag_if_supported("-Wl,--icf=safe");
 
+    if cfg!(feature = "fat-lto") {
+        cfg.flag_if_supported("-flto");
+    } else if cfg!(feature = "thin-lto") {
+        if cfg.is_flag_supported("-flto=thin").unwrap_or(false) {
+            cfg.flag("-flto=thin");
+        } else {
+            cfg.flag_if_supported("-flto");
+        }
+    }
```

Comment:
I've asked at https://rust-lang.zulipchat.com/#narrow/channel/246057-t-cargo/topic/read.20lto.20config.20from.20build.20script if there is a way to read the LTO state from cargo without having to use feature flags. In any case this won't actually LTO between bzip2 and any rust code. That is only possible when using the rust port of bzip2 rather than the C original or by using `-Clinker-plugin-lto` and making sure that the exact right clang version is used to build the C code.

Reply:
Turns out the cc crate itself nowadays handles `-Clto` and `-Copt-level` already, so the second and third commit can be dropped. Only the first and fourth commit (which should be squashed) are still necessary.